website/blog/deployers.md

```diff
@@ -0,0 +1,610 @@
+illustrated below.
+
+<figure>
+  <img src="deployers/assets/overview.svg" alt="An architecture diagram of a deployer. An app with components A, B, and C is compiled into a binary which is deployed by a deployer across three weavelets.">
+</figure>
+
+In this blog post, we provide a high-level overview of weavelets, deployers, and
+envelopes. Then, we get down to the nitty-gritty of exactly how they work by
+implementing a multiprocess deployer completely from scratch.
+
+## Weavelets
+
+To understand deployers, we must first understand weavelets. A Service Weaver
+application is compiled into a single executable binary. The Service Weaver
+libraries linked into the binary include a small agent called a **weavelet**,
+which is created when you call [weaver.Init][Init]. A weavelet's main
+responsibility is to start and manage a set of [components][].
+
+When a Service Weaver application is deployed, there isn't just one weavelet. If
+there were, Service Weaver applications wouldn't be very distributed. Rather,
+deployers run your binary multiple times&mdash;in different processes across
+different machines&mdash;to launch multiple weavelets which work together to
+execute your distributed application.
+
+Every weavelet hosts a potentially different set of components. Because
+components are replicated, a component may be hosted by multiple weavelets. For
+example, consider an application with components `A`, `B`, and `C`. An
+example deployment consisting of three weavelets is shown in the figure below.
+Weavelet 1 hosts components `A` and `B`; weavelet 2 hosts components `B` and
+`C`, and weavelet 3 hosts component `C`.
+
+<figure>
+  <img src="deployers/assets/weavelets.svg" alt="Three weavelets hosting different subsets of components A, B, and C. Each weavelet has its own network address.">
+</figure>
+
+You'll also notice that every weavelet has a unique network address. Weavelets
+use these addresses to execute remote methods calls. For example, imagine
+component `A` on weavelet 1 in the figure above wants to call a method on
+component `C`. Weavelet 1 will contact either weavelet 2 on address 2.2.2.2 or
+weavelet 3 on address 3.3.3.3 to execute the method.
+
+## Deployers
+
+A deployer distributes a Service Weaver application by launching and managing a
+set of weavelets. Managing weavelets involves three main responsibilities
+related to (1) components, (2) listeners, and (3) telemetry.
+
+1. **Components**. A deployer starts weavelets and tells them which components
+   to host. A deployer also ensures that weavelets know the addresses of other
+   weavelets. If a deployer starts a new weavelet, for example, the deployer
+   notifies all other weavelets of its existence, including its address and the
+   components its hosting. Conversely, if a deployer detects that a weavelet has
+   failed, the deployer notifies all other weavelets of its failure.
+
+2. **Listeners**. When a component calls the [Instance.Listener][Listener]
```

Comment:
How about:

When a component wants to serve external traffic, it requests a network listener. The deployer picks an address ...